src/app/services/config/config.service.ts

```diff
@@ -19,30 +22,52 @@ export const assetRoot = "/assets";
  */
 @Injectable()
 export class ConfigService {
+  private _validConfig: boolean;
   private _config: Configuration;
+  private http: HttpClient;
 
   public constructor(
     private notification: ToastrService,
     private theme: ThemeService,
+    handler: HttpBackend,
     @Inject(IS_SERVER_PLATFORM) private isServer: boolean
   ) {
-    if (!isConfiguration(environment, this.isServer)) {
-      console.error("Detected invalid environment.");
-      this.notification.error(
-        "The website is not configured correctly. Try coming back at another time.",
-        "Unrecoverable Error",
-        {
-          closeButton: false,
-          disableTimeOut: true,
-          tapToDismiss: false,
-          positionClass: "toast-center-center",
-        }
-      );
+    // This is to bypass the interceptor and prevent circular dependencies
+    // (interceptor requires API_ROOT)
+    // https://stackoverflow.com/questions/57850927/angular-app-initializer-circular-dependencies-at-runtime
+    this.http = new HttpClient(handler);
+  }
+
+  public async init(defaultConfig?: Promise<Configuration>): Promise<void> {
+    const embedGoogleServicesIfValid = async () => {
+      // Only insert if valid config, and not SSR
+      if (this.validConfig && !this.isServer) {
+        await embedGoogleMaps(this.keys.googleMaps);
+        await embedGoogleAnalytics(this.keys.googleAnalytics.trackingId);
+      }
+    };
+
+    if (defaultConfig) {
+      this.setConfig(await defaultConfig);
+      await embedGoogleServicesIfValid();
       return;
     }
 
-    this._config = new Proxy(environment, {});
-    this.theme.setTheme(this.settings.theme ?? {});
+    return firstValueFrom(
+      this.http.get("assets/environment.json").pipe(
+        retry({ count: 5, delay: 1000 }),
```

Comment:
why such a large delay?

Reply:
I've changed this down to a 0 second delay, but I wonder if we still need to retry requests. I'm not sure if this will help any users

Reply:
well some delay is useful... 250ms maybe?

did you add the retry for a reason?